nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/test/java/org/apache/nifi/processors/mongodb/RunMongoAggregationTest.java

```diff
@@ -0,0 +1,146 @@
+        runner = TestRunners.newTestRunner(RunMongoAggregation.class);
+        runner.setVariable("uri", MONGO_URI);
+        runner.setVariable("db", DB_NAME);
+        runner.setVariable("collection", COLLECTION_NAME);
+        runner.setProperty(AbstractMongoProcessor.URI, "${uri}");
+        runner.setProperty(AbstractMongoProcessor.DATABASE_NAME, "${db}");
+        runner.setProperty(AbstractMongoProcessor.COLLECTION_NAME, "${collection}");
+
+        mongoClient = new MongoClient(new MongoClientURI(MONGO_URI));
+
+        MongoCollection<Document> collection = mongoClient.getDatabase(DB_NAME).getCollection(COLLECTION_NAME);
+        String[] values = new String[] { "a", "b", "c" };
+        mappings = new HashMap<>();
+
+        for (int x = 0; x < values.length; x++) {
+            for (int y = 0; y < x + 2; y++) {
+                Document doc = new Document().append("val", values[x]);
+                collection.insertOne(doc);
+            }
+            mappings.put(values[x], x + 2);
+        }
+    }
+
+    @After
+    public void teardown() {
+        runner = null;
+
+        mongoClient.getDatabase(DB_NAME).drop();
+    }
+
+    @Test
+    public void testAggregation() throws Exception {
+        runner.setProperty(RunMongoAggregation.QUERY, "[\n" +
+                "    {\n" +
+                "        \"$project\": {\n" +
+                "            \"_id\": 0,\n" +
+                "            \"val\": 1\n" +
+                "        }\n" +
+                "    },\n" +
+                "    {\n" +
+                "        \"$group\": {\n" +
+                "            \"_id\": \"$val\",\n" +
+                "            \"doc_count\": {\n" +
+                "                \"$sum\": 1\n" +
+                "            }\n" +
+                "        }\n" +
+                "    }\n" +
+                "]");
+        runner.enqueue("test");
+        runner.run(1, true, true);
+
+        evaluateRunner();
+    }
+
+    @Test
+    public void testExpressionLanguageSupport() throws Exception {
+        runner.setVariable("fieldName", "$val");
+        runner.setProperty(RunMongoAggregation.QUERY, "[\n" +
+                "    {\n" +
+                "        \"$project\": {\n" +
+                "            \"_id\": 0,\n" +
+                "            \"val\": 1\n" +
+                "        }\n" +
+                "    },\n" +
+                "    {\n" +
+                "        \"$group\": {\n" +
+                "            \"_id\": \"${fieldName}\",\n" +
+                "            \"doc_count\": {\n" +
+                "                \"$sum\": 1\n" +
+                "            }\n" +
+                "        }\n" +
+                "    }\n" +
+                "]");
+
+        runner.setValidateExpressionUsage(true);
+        runner.enqueue("test");
+        runner.run(1, true, true);
+        evaluateRunner();
+    }
+
+    private void evaluateRunner() throws IOException {
```

Comment:
Please add unit test(s) around various failure conditions where prudent (perhaps with a mock on the connection or whatever).